File: src/main/java/shopping/cart/domain/entity/Product.java
package shopping.cart.domain.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue
    private Long id;
    private String name;
    private String imageUuidFileName;
    private int price;

    public Product() {
    }

    public Product(final Long id, final String name, final String imageUuidFileName,
        final int price) {
        this.id = id;
        this.name = name;
        this.imageUuidFileName = imageUuidFileName;
        this.price = price;
    }

    public Product(final String name, final String imageUuidFileName, final int price) {
        this(null, name, imageUuidFileName, price);
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImageUuidFileName() {
        return imageUuidFileName;
    }

    public int getPrice() {
        return price;
    }
}

File: src/main/kotlin/shopping/product/entity/Product.kt
package shopping.product.entity

import jakarta.persistence.*

@Entity
data class Product(
    @Column(nullable = false)
    var name: String,
    @Column(nullable = false)
    var price: Int,
    @Column(nullable = false)
    var imageUrl: String,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
)
